homeassistant/core.py

```diff
@@ -637,30 +671,27 @@ async def async_block_till_done(self) -> None:
         """Block until all pending work is done."""
         # To flush out any call_soon_threadsafe
         await asyncio.sleep(0)
+        await asyncio.sleep(0)
```

Comment:
We used to do this by always entering the while loop, because we only checked if there were done tasks to be cleared afterwards.